pkg/services/profiles/update.go

```diff
@@ -0,0 +1,146 @@
+package profiles
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/weaveworks/weave-gitops/pkg/git"
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders"
+	"github.com/weaveworks/weave-gitops/pkg/helm"
+	"github.com/weaveworks/weave-gitops/pkg/models"
+
+	"github.com/fluxcd/go-git-providers/gitprovider"
+	helmv2beta1 "github.com/fluxcd/helm-controller/api/v2beta1"
+	"github.com/google/uuid"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const UpdateCommitMessage = "Update Profile manifests"
+
+type UpdateOptions struct {
+	Name         string
+	Cluster      string
+	ConfigRepo   string
+	Version      string
+	ProfilesPort string
+	Namespace    string
+	Kubeconfig   string
+	AutoMerge    bool
+}
+
+// Update updates an installed profile
+func (s *ProfilesSvc) Update(ctx context.Context, gitProvider gitproviders.GitProvider, opts UpdateOptions) error {
+	configRepoURL, err := gitproviders.NewRepoURL(opts.ConfigRepo)
+	if err != nil {
+		return fmt.Errorf("failed to parse url: %w", err)
+	}
+
+	repoExists, err := gitProvider.RepositoryExists(ctx, configRepoURL)
+	if err != nil {
+		return fmt.Errorf("failed to check whether repository exists: %w", err)
+	} else if !repoExists {
+		return fmt.Errorf("repository %q could not be found", configRepoURL)
+	}
+
+	defaultBranch, err := gitProvider.GetDefaultBranch(ctx, configRepoURL)
+	if err != nil {
+		return fmt.Errorf("failed to get default branch: %w", err)
+	}
+
+	helmRepo, version, err := s.discoverHelmRepository(ctx, GetOptions{
+		Name:      opts.Name,
+		Version:   opts.Version,
+		Cluster:   opts.Cluster,
+		Namespace: opts.Namespace,
+		Port:      opts.ProfilesPort,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to discover HelmRepository: %w", err)
+	}
+
+	opts.Version = version
+
+	files, err := gitProvider.GetRepoDirFiles(ctx, configRepoURL, git.GetSystemPath(opts.Cluster), defaultBranch)
+	if err != nil {
+		return fmt.Errorf("failed to get files in '%s' of config repository %q: %s", git.GetSystemPath(opts.Cluster), configRepoURL, err)
+	}
+
+	fileContent := getGitCommitFileContent(files, git.GetProfilesPath(opts.Cluster, models.WegoProfilesPath))
+	if fileContent == "" {
+		return fmt.Errorf("failed to find installed profiles in '%s' of config repo %q", git.GetProfilesPath(opts.Cluster, models.WegoProfilesPath), configRepoURL)
+	}
+
+	content, err := updateHelmRelease(helmRepo, fileContent, opts.Name, opts.Version, opts.Cluster, opts.Namespace)
+	if err != nil {
+		return fmt.Errorf("failed to update HelmRelease for profile '%s' in %s: %w", opts.Name, models.WegoProfilesPath, err)
+	}
```

Comment:
This is a bit weird here. The `getGitCommitFileContent` is not used anywhere else but `updateHelmRelease`. But also, `updateHelmRelease` doesn't use `helmRepo` at all. It's unused. :D I think something can be shuffled around here.

Reply:
Which is super weird, because then the result of this:

```go
helmRepo, version, err := s.discoverHelmRepository(ctx, GetOptions{
		Name:      opts.Name,
		Version:   opts.Version,
		Cluster:   opts.Cluster,
		Namespace: opts.Namespace,
		Port:      opts.ProfilesPort,
	})
```

Has no actual result other than maybe version being set on options. 🤔